src/effect.js

```diff
@@ -0,0 +1,112 @@
+define(function (require) {
+	'use strict';
+
+	var p5sound = require('master');
+
+	p5.Effect = function(type) {
+		//everything points to the same audiocontext
+		this.type = type;
+		
+		this.ac = p5sound.audiocontext;
+
+		//wrap each node in gain nodes
+		this.input = this.ac.createGain();
+		this.output = this.ac.createGain();
+
+		this.dry = this.ac.createGain();
+		this.wet = this.ac.createGain();
+
+
+		// this.input.connect(this.output);
+		//connect input to wet and dry gain nodes
+		// this.input.connect(this.dry);
+		// this.input.connect(this.wet);
+
+		// //connect wet and dry gain nodes to output
+		// this.wet.connect(this.output);
+		// this.dry.connect(this.output);
+
+		this.input.connect(this.output);
+
+		//connects to audiocontext destination
+		this.connect();
+
+		if (type) {
+			// determine the type of effect?
+			console.log(type);
+		}
```

Comment:
same comment as above—I think this belongs in the Filter constructor for now at least.